backend/backend.go

```diff
@@ -683,3 +685,62 @@ func (backend *Backend) HandleURI(uri string) {
 		backend.log.Warningf("Unknown URI scheme: %s", uri)
 	}
 }
+
+// PocketWidgetSignAddress returns an unused address and makes the user sign a message to prove ownership.
+func (backend *Backend) PocketWidgetSignAddress(code accounts.Code, msg string, format string) (string, string) {
+	log := backend.log.WithField("accountCode", code)
+	var account accounts.Interface
+	for _, acct := range backend.accounts {
+		if acct.Config().Code == code {
+			account = acct
+			break
+		}
+	}
+	if account == nil {
+		log.Error("Pocket Widget: could not find account")
+		return "", ""
+	}
+	if err := account.Initialize(); err != nil {
+		log.
+			WithError(err).
+			WithField("code", account.Config().Code).
+			Error("Pocket Widget: could not initialize account")
+		return "", ""
+	}
+	if account.Coin().Code() != coinpkg.CodeBTC {
+		log.
+			WithField("code", account.Config().Code).
+			Error("Pocket Widget: coin not supported")
+		return "", ""
+	}
+
+	unused := account.GetUnusedReceiveAddresses()
+	// Use the format hint to get a compatible address
+	expectedScriptType, ok := aoppBTCScriptTypeMap[format]
+	if !ok {
+		log.Errorf("Pocket Widget: Unknown format param %s", format)
+		return "", ""
+	}
+	signingConfigIdx := account.Config().SigningConfigurations.FindScriptType(expectedScriptType)
+	if signingConfigIdx == -1 {
+		log.Errorf("Pocket Widget: Unknown format param %s", format)
+		return "", ""
+	}
+	addr := unused[signingConfigIdx].Addresses[0]
+
+	sig, err := backend.keystore.SignBTCMessage(
+		[]byte(msg),
+		addr.AbsoluteKeypath(),
+		account.Config().SigningConfigurations[signingConfigIdx].ScriptType(),
+	)
+	if err != nil {
+		if firmware.IsErrorAbort(err) {
```

Comment:
this would definitely be helpful to propagate. maybe an 'user aborted' flag/msg should even be accepted by pocket? we should ask them about it.

Reply:
Added an `abort` flag in the `Response` struct. Atm the frontend just shows an alert, but we could send a message to the widget for this specific case, if we decide so.